src/frontend/src/optimizer/plan_node/logical_change_log.rs

```diff
@@ -0,0 +1,157 @@
+        let change_log = Self::new(input, self.core.need_op, true);
+        if self.core.need_op {
+            let mut output_vec = input_col_change.to_parts().0.to_vec();
+            let len = input_col_change.to_parts().1;
+            output_vec.push(Some(len));
+            let out_col_change = ColIndexMapping::new(output_vec, len + 1);
+            (change_log, out_col_change)
+        } else {
+            (change_log, input_col_change)
+        }
+    }
+}
+
+impl_plan_tree_node_for_unary! {LogicalChangeLog}
+impl_distill_by_unit!(LogicalChangeLog, core, "LogicalChangeLog");
+
+impl ExprRewritable for LogicalChangeLog {}
+
+impl ExprVisitable for LogicalChangeLog {}
+
+impl PredicatePushdown for LogicalChangeLog {
+    fn predicate_pushdown(
+        &self,
+        predicate: Condition,
+        ctx: &mut super::PredicatePushdownContext,
+    ) -> PlanRef {
+        gen_filter_and_pushdown(self, predicate, Condition::true_cond(), ctx)
+    }
+}
+
+impl ColPrunable for LogicalChangeLog {
+    fn prune_col(&self, required_cols: &[usize], ctx: &mut ColumnPruningContext) -> PlanRef {
+        let fields = self.schema().fields();
+        let mut need_op = false;
+        let mut need_change_log_row_id = false;
+        let new_required_cols: Vec<_> = required_cols
+            .iter()
+            .filter_map(|a| {
+                if let Some(f) = fields.get(*a) {
+                    if f.name == CHANGE_LOG_OP {
+                        need_op = true;
+                        None
+                    } else if f.name == _CHANGE_LOG_ROW_ID {
+                        need_change_log_row_id = true;
+                        None
+                    } else {
+                        Some(*a)
+                    }
+                } else {
+                    Some(*a)
+                }
+            })
+            .collect();
+
+        let new_input = self.input().prune_col(&new_required_cols, ctx);
+        Self::new(new_input, need_op, need_change_log_row_id).into()
+    }
+}
+
+impl ToBatch for LogicalChangeLog {
+    fn to_batch(&self) -> Result<PlanRef> {
+        Err(BindError("With changelog cte only support with create mv/sink".to_string()).into())
+    }
+}
+
+impl ToStream for LogicalChangeLog {
+    fn to_stream(&self, ctx: &mut ToStreamContext) -> Result<PlanRef> {
+        let new_input = self.input().to_stream(ctx)?;
+
+        let mut new_logical = self.core.clone();
+        new_logical.input = new_input;
+        let plan = StreamChangeLog::new(new_logical).into();
+        let row_id_index = self.schema().fields().len() - 1;
+        let plan = StreamRowIdGen::new_with_dist(
+            plan,
+            row_id_index,
+            Distribution::HashShard(vec![row_id_index]),
+        )
+        .into();
+
+        Ok(plan)
+    }
+
+    fn logical_rewrite_for_stream(
+        &self,
+        ctx: &mut RewriteStreamContext,
+    ) -> Result<(PlanRef, ColIndexMapping)> {
+        ctx.set_with_stream_key(false);
```

Comment:
I think we need to reset the `with_stream_key` after we rewrite the input. Otherwise, other unrelated table in the same query will be affected by this flag too.